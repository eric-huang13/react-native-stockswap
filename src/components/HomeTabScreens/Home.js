import React, {Component, useState, useEffect, useRef} from 'react';
import UserPosts from '../UserPost/UserPosts';
import StockTicker from '../HomeTabComponents/StockTicker';
import {connect} from 'react-redux';
import {
  Button,
  Modal,
  SafeAreaView,
  Text,
  ScrollView,
  StyleSheet,
} from 'react-native';
import {fetchTickers} from '../../actions/marketMovers';
import ReportModal from '../HomeTabComponents/ReportModal';
import ShareToModal from '../HomeTabComponents/ShareToModal';
import {Logout} from 'actions/user';
import MoreBox from '../HomeTabComponents/MoreBox';
import {PortfolioAccounts} from '../../actions/profile';
import {
  EDIT_POST,
  REMOVE_POST,
  REPOST,
  COPY_LINK,
  SHARE_POST,
  TURN_ON_NOTIFICATIONS,
  REPORT,
  myPostsOptions,
  otherPostsOptions,
} from './constants';

const HomeScreen = ({
  navigation,
  fetchTickers,
  isLoggedIn,
  LogoutUser,
  posts,
  comments,
  reply,
  userAccount,
  PortfolioAccounts
}) => {
  const [reportModalState, setReportModalState] = useState(false);
  const [shareModalState, setShareModalState] = useState(false);
  const [optionsModalState, setOptionsModalState] = useState(false);
  const [optionsState, setOptionsState] = useState([]);
  const accountId = userAccount.id;

  useEffect(() => {
    fetchTickers && fetchTickers();
    PortfolioAccounts();
  }, []);

  const reportModal = () => {
    setReportModalState(true);
  };

  const shareModal = () => {
    setShareModalState(true);
  };

  const optionModal = (post) => {
    setOptionsModalState(true);
    if (post?.userId == accountId) {
      setOptionsState(myPostsOptions);
    } else {
      setOptionsState(otherPostsOptions);
    }
  };

  const onSelectOption = (action) => {
    switch (action) {
      case EDIT_POST:
        break;
      case REMOVE_POST:
        break;
      case REPOST:
        break;
      case COPY_LINK:
        break;
      case SHARE_POST:
        shareModal();
        break;
      case TURN_ON_NOTIFICATIONS:
        break;
      case REPORT:
        reportModal();
        break;
      default:
        console.log('Not a valid action');
    }
  };

  return (
    <SafeAreaView style={style.mainContainer}>
      <ScrollView>
        <StockTicker />
        {posts.map((post) => (
          <UserPosts
            key={post.id}
            post={post}
            navigation={navigation}
            comments={comments}
            reply={reply}
            userAccount={userAccount}
            reportModal={reportModal}
            shareModal={shareModal}
            optionModal={optionModal}
          />
        ))}
      </ScrollView>
      <Modal
        transparent={true}
        visible={reportModalState}
        animationType="slide">
        <ReportModal
          reportModal={reportModal}
          onClose={() => setReportModalState(false)}
        />
      </Modal>
      <Modal transparent={true} visible={shareModalState} animationType="slide">
        <ShareToModal
          shareModal={setShareModalState}
          onClose={() => setShareModalState(false)}
        />
      </Modal>
      <Modal
        transparent={true}
        visible={optionsModalState}
        animationType="slide">
        <MoreBox
          options={optionsState}
          onClose={() => setOptionsModalState(false)}
          onSelect={onSelectOption}
        />
      </Modal>
    </SafeAreaView>
  );
};

const mapStateToProps = (state) => {
  return {
    isLoggedIn: state.user.isLoggedIn,
    posts: state.posts.posts,
    comments: state.posts.comments,
    reply: state.posts.reply,
    userData: state.user.userData,
    userAccount: state.user.userFakeData,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    LogoutUser: (userCredentials) => dispatch(Logout(userCredentials)),
    fetchTickers: () => dispatch(fetchTickers()),
    GetProfileImage: (token, id) => dispatch(GetProfileImage(token, id)),
    RefreshToken: (token) => dispatch(RefreshToken(token)),
    PortfolioAccounts: () => dispatch(PortfolioAccounts()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

const style = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: '#2a334a',
  },
});
