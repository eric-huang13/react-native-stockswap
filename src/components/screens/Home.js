import React, {Component} from 'react';
import UserPosts from './UserPosts';
import StockTicker from './StockTicker';
import {connect} from 'react-redux';
import {Button, SafeAreaView, Text, ScrollView, StyleSheet} from 'react-native';
import AsyncStorage from "@react-native-async-storage/async-storage";


import {Logout} from 'actions/user';

class HomeScreen extends Component {

  constructor(props) {
    super(props);

    this.state = {
      token: "",
   
      
    };
  }
  render() {
    const {isLoggedIn, LogoutUser, posts, comments, reply, userData} = this.props;
    console.log(userData,"USERDATA IN HOME")
    
     const getData = async () => {
    try {
      const value = await AsyncStorage.getItem("token");
      if (value !== null) {
        this.setState({
          token: value,
        });
      }
    } catch (error) {
      console.log('error')
      // error reading value
    }
  };
    // const token = AsyncStorage.getItem("userToken")
    // console.log(token, "TOKEN")
    return (
      <SafeAreaView style={style.mainContainer}>
        <ScrollView>
          <StockTicker />
          <Text>{userData.message}</Text>

          {posts.map((post) => (
            <UserPosts
              key={post.id}
              post={post}
              navigation={this.props.navigation}
              comments={comments}
              reply={reply}
            />
          ))}
          <Text>Is User Logged in: {'' + isLoggedIn} </Text>
          <Button title="Logout Button" onPress={() => LogoutUser()} />
          <Text>{this.state.token}</Text>
          <Button title="Get Token" onPress={() => getData()} />

        </ScrollView>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isLoggedIn: state.user.isLoggedIn,
    posts: state.posts.posts,
    comments: state.posts.comments,
    reply: state.posts.reply,
    userData: state.user.userData,

  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    LogoutUser: (userCredentials) => dispatch(Logout(userCredentials)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

const style = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: '#2a334a',
  },
});
