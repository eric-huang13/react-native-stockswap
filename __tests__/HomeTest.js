import React from 'react';
import Home from '../src/components/HomeTabScreens/Home';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Home posts={[]}/>);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });