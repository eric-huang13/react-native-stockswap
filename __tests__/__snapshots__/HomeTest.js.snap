// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
    }
  }
>
  <ScrollView>
    <StockTicker />
    <Text>
      Is User Logged in: 
      undefined
       
    </Text>
    <Button
      onPress={[Function]}
      title="Logout Button"
    />
  </ScrollView>
</ForwardRef(SafeAreaView)>
`;
